### PR TITLE
add store url to bugsnag metadata, DRY - move adding API metadata and…

### PR DESCRIPTION
**Pull request by danac-gs**
… headers to Hook Helper

### CODE REVIEW

**Comment by YevhenShBolt** [comment]
Just want to clarify.

For me it is in vendor folder, it looks like:
- `vendor/bugsnag`
- `vendor/guzzle`

How did you get it in `lib/internal` folder?

**Comment by danac-gs** [comment]
@YevhenShBolt It is in vendor as it should always be if installed via composer. This is added to support composerless install (requested by Capillus in the past). It was commented out but I uncommented it in order to not have to edit/change the code if we would have the similar request in the future. It does not break anything, can go like that. For composerless installs, on demand, we will provide the additional contents to put in the lib folder. It is not a part of the plugin core.

File: Helper/Hook.php [hunk 1]
```diff
@@ -154,4 +168,31 @@ public function verifyWebhook()
...
     }
+
+    /**
+     * Set bugsnag metadata bolt_trace_id
+     */
+    public function setMetadata()
+    {
+        if ($bolt_trace_id = $this->request->getHeader(ConfigHelper::BOLT_TRACE_ID_HEADER)) {
+            $this->bugsnag->registerCallback(function ($report) use ($bolt_trace_id) {
+                $report->setMetaData([
```
Comment (daisy1754) [inline Helper/Hook.php]: shouldn't this addMetaData so it won't override existing metadata with store URL?

File: Helper/Hook.php [hunk 2]
```diff
@@ -154,4 +168,31 @@ public function verifyWebhook()
             throw new WebapiException(__('Precondition Failed'), 6001, 412);
         }
     }
+
+    /**
+     * Set bugsnag metadata bolt_trace_id
+     */
+    public function setMetadata()
```
Comment (daisy1754) [inline Helper/Hook.php]: maybe setCommonMetaData?

Also nit: inconsistent capitalization about Metadata vs MetaData

File: Helper/Hook.php (hunk 1 above)
Comment (danac-gs) [reply Helper/Hook.php]: @daisy1754 In fact it is the opposite. setMetaData by default uses array_merge_recursive which keeps multiple values in an array.

```php
    public function setMetaData(array $metaData, $merge = true)
    {
        $this->metaData = $merge ? array_merge_recursive($this->metaData, $metaData) : $metaData;

        return $this;
    }
    public function addMetaData(array $metadata)
    {
        $this->metaData = array_replace_recursive($this->metaData, $metadata);
        $this->metaData = $this->removeNullElements($this->metaData);

        return $this;
    }
```

File: Helper/Hook.php (hunk 2 above)
Comment (danac-gs) [reply Helper/Hook.php]: @daisy1754 fixed. Helper::Bugsnag::addCommonMetadata() as well.

**Pull request merged**